### PR TITLE
Add discriminator property to generated typeables. Fix #113

### LINKED ISSUE #113: Generated typeables should contain discriminator field if specified
Example from https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/:

```
paths:
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/Cat'
                - $ref: '#/components/schemas/Dog'
              discriminator:
                propertyName: pet_type
      responses:
        '200':
          description: Updated
components:
  schemas:
    Pet:
      type: object
      required:
        - pet_type
      properties:
        pet_type:
          type: string
      discriminator:
        propertyName: pet_type
    Dog:     # "Dog" is a value for the pet_type property (the discriminator value)
      allOf: # Combines the main `Pet` schema with `Dog`-specific properties 
        - $ref: '#/components/schemas/Pet'
        - type: object
          # all other properties specific to a `Dog`
          properties:
            bark:
              type: boolean
            breed:
              type: string
              enum: [Dingo, Husky, Retriever, Shepherd]
    Cat:     # "Cat" is a value for the pet_type property (the discriminator value)
      allOf: # Combines the main `Pet` schema with `Cat`-specific properties 
        - $ref: '#/components/schemas/Pet'
        - type: object
          # all other properties specific to a `Cat`
          properties:
            hunts:
              type: boolean
            age:
              type: integer
```

### PR DESCRIPTION
**Pull request by danslapman**
Object schemas in oneOf/allOf must explicitly contains property named exactly as specified discriminator. This PR should fix SwaggerUI respresentation of sealed traits with discriminator property configured.

Reference: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#discriminator-object

### CODE REVIEW

**Comment by danslapman** [comment]
2.12 build failed due to a strange trouble with maven:

```
[error] lmcoursier.internal.shaded.coursier.error.FetchError$DownloadingArtifacts: Error fetching artifacts:
[error] https://repo1.maven.org/maven2/com/twitter/util-security_2.12/20.8.0/util-security_2.12-20.8.0.jar: wrong checksum: /home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/twitter/util-security_2.12/20.8.0/util-security_2.12-20.8.0.jar (expected SHA-1 c3951e4999498c95e659bdccf1b01ddebd29b251 in /home/runner/.cache/coursier/v1/https/repo1.maven.org/maven2/com/twitter/util-security_2.12/20.8.0/.util-security_2.12-20.8.0.jar__sha1, got e90666a3e5fd9454606988954d0899ec8d6ae6d6)
```

**Comment by Odomontois** [comment]
I've merged the upgrate of twitter lib, try rebase on fresh master

**Comment by danslapman** [comment]
Rebased

**Comment by Odomontois** [comment]
That looks like it could break existing code.
If a user has global "Configuration" with "discriminator" enabled for some modules, only their sealed traits are supplied with a required discriminator field, from now one plain records will have discriminator as well.
I propose to add some boolean config field, i.e. `forceDiscriminator` or `productDiscriminator` with defalt value `false` to keep current behaviour.

**Review by Odomontois — APPROVED** [review]

File: modules/finagle-env/src/test/scala/ru/tinkoff/tschema/finagle/AuthorizationSpec.scala [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package ru.tinkoff.tschema.finagle
 
+import scala.language.reflectiveCalls
```
Comment (Odomontois) [inline modules/finagle-env/src/test/scala/ru/tinkoff/tschema/finagle/AuthorizationSpec.scala:3]: Why is this needed?

Comment (danslapman) [reply modules/finagle-env/src/test/scala/ru/tinkoff/tschema/finagle/AuthorizationSpec.scala:3]: Just to silence scalac warning
